### ISSUE TITLE
Is there a way to cause the REDIS_ERR_EOF not the REDIS_ERR_IO?

### ISSUE THREAD

**Issue by ruokuanwu** [redis/hiredis#853]
HI, I want to simulate a network fault to test the my application whether switches automatically the IP of redis service. The redis service was installed in my virtualbox which applied two host-only virtual network card, `enp0s3` and `enp0s5`. Actually, when I use the command `ifconfig enp0s3 down`, I found that the `err` of `redisContext` is  `REDIS_ERR_IO` which takes too much time to try to connect the died IP seemly. However, when I shutdown my virtual box, the `err` becomes `REDIS_ERR_EOF ` which returns fastly.
 I want the application to return fastly. So ,Is there a way to cause the `REDIS_ERR_EOF` not the `REDIS_ERR_IO`?

**Comment by ruokuanwu** [comment]
Now, I can use such commands to simulate the situation I want.
`iptables -t nat -A PREROUTING -d 192.168.84.101 -p tcp --dport 16379 -j REDIRECT --to-ports 6379`
In my application, I used the `192.168.84.101:16379`. 

But another question, When I use `redis-cli -h 192.168.84.101 -p 16379` in the virtual box, it returns `Could not connect to Redis at 192.168.84.101:16379: Connection refused`. Now I have no idea about this.

These are my iptables-rules :
```
Chain PREROUTING (policy ACCEPT 100 packets, 14934 bytes)
 pkts bytes target     prot opt in     out     source               destination
  128  6656 REDIRECT   tcp  --  *      *       0.0.0.0/0            192.168.84.101       tcp dpt:16379 redir ports 6379

Chain INPUT (policy ACCEPT 228 packets, 21590 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain OUTPUT (policy ACCEPT 77 packets, 6514 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain POSTROUTING (policy ACCEPT 77 packets, 6514 bytes)
 pkts bytes target     prot opt in     out     source               destination

```
and the redis.conf
```
...
bind 192.168.56.101 192.168.45.101 192.168.84.101  127.0.0.1
protected-mode no
port 6379
...
```

In conclusion, I want to find out the reason why redis returns `Could not connect to Redis at 192.168.84.101:16379` in my virtualbox.
